files/Aboot/boot0.j2

```diff
@@ -650,6 +651,32 @@ write_default_cmdline() {
     cmdline_add "$delimiter"
 }
 
+write_cmdline() {
+    # use extra parameters from kernel-params hook if the file exists
+    if [ -f "$target_path/$kernel_params" ]; then
+        if $secureboot && $debug; then
```

Comment:
>if $secureboot && $debug; then [](start = 8, length = 30)

If secureboot and not debug, you will not use extra parameter. This is surprising different than the case of secureboot and debug.

Reply:
That's correct.
The `$target_path/$kernel_params` file which is `/host/kernel-params` allows the user to inject more kernel parameters on the cmdline. It's especially useful during development to figure out which set of parameters should be used on a platform. But it can also be used to change some runtime behavior. Unfortunately some could have unfortunate behavior and end up compromising a system. To remove the risk of an attacker injecting harmful parameter we disable this feature in secureboot and print a message. It is still possible to boot the box in debug mode and inject parameters but that will show at runtime.
If you prefer I can disable this mechanism for secureboot entirely, including `secureboot && debug`.